### ISSUE TITLE
Continuous integration

### ISSUE THREAD

**Issue by samuelpowell** [toastpp/toastpp#11]
Tracking issue for continuous integration
- [x] Linux, OS X back end library build on Travis CI (see #1)
- [x] Windows library build on AppVeyor (see #36)
- [x] Python interface builds (could be enabled on Travis?)
- [ ] Extract nightly binaries?
- [ ] MATLAB interface builds

**Comment by mschweiger** [comment]
@samuelpowell: would it be possible to run two separate build tests on linux:

./configure; make numerics; make toast

and

./configure --enable-threads --enable-fdot; make numerics; make toast

or alternatively, replace the first version with the second one (since that is the configuration generally used for building releases).

**Comment by samuelpowell** [comment]
@mschweiger should be fixed by #21, I will pull the changes into the parameter branch to get CI report.

**Comment by samuelpowell** [comment]
@mschweiger could you have a look [here](https://www.appveyor.com/docs/build-phase) and work out what sequence of commands will work to build the Windows binaries? If you let me know I'll put together the configuration file and see how far we get...

**Comment by mschweiger** [comment]
@samuelpowell ok, I've got the VS2015 toolchain working now in the master branch. The commands for building the Windows binaries should now simply be (starting from the toast home directory):

cd win\VS2015
msbuild toast.sln /p:Platform="x64" /p:Configuration="Release_fdot"

This does require Visual Studio 2015 (Toast won't build on VS2010 to VS2013). I don't quite know how Travis and AppVeyor are connected and how you integrate the VS build commands into travis.yml. How do I proceed?

**Comment by samuelpowell** [comment]
Travis and AppVeyor are completely separate CI systems. I'm having a go at putting together an appveyor.yml configuration file in #36. Let's see how it goes!

**Comment by samuelpowell** [comment]
@mschweiger regarding CI binaries, we can have these pushed to a variety of places (e.g., S3, or a CS FTP site, see a full list [here](https://docs.travis-ci.com/user/deployment/script/), and [here](https://www.appveyor.com/docs/deployment). Let me know your thoughts.

**Comment by samuelpowell** [comment]
Python backend CI enabled on Linux and Mac in #46, Windows pending. Ref #55.

**Comment by samuelpowell** [comment]
@mschweiger is Octave an option for testing the MEX interface?

**Comment by mschweiger** [comment]
@samuelpowell I have never tried to build the mex files or test the Matlab interface with octave. Might be worth a try.

Another idea: if we could set up a CI system that integrates with github but allows the builds to take place on a local machine rather than a cloud server then maybe we could circumvent the Matlab installation problem. I had a brief look and it seems that there is something called Jenkins that might allow to do that. Have you come across Jenkins? I'm a bit out of my depth trying to figure out how to set this up.

**Comment by samuelpowell** [comment]
Yes, the latter idea is probably worth pursuing. UCL RITS maintain in-house CI based on Jenkins, which means we could use the MATLAB site-license and their servers. I'm not sure about the terms of service. Perhaps we should contact them to discuss.
